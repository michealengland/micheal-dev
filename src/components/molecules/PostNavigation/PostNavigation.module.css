.postNavigation {
  @apply bg-primary-purple2 text-nuetral-white p-4;

  & .inner {
    @apply flex justify-between max-w-3xl mx-auto;
  }

  & li {
    @apply list-none;
  }

  & a {
    @apply text-nuetral-white no-underline;

    &:hover,
    &:focus {
      @apply text-secondary-pink underline;
    }
  }
}
