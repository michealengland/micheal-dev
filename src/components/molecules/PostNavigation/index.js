import React from 'react'
import TextLink from '@components/atoms/TextLink'
import PropTypes from 'prop-types'
import styles from './PostNavigation.module.css'

export default function PostNavigation({nextLabel, nextTo, prevLabel, prevTo}) {
  return (
    <ul className={styles.postNavigation}>
      <li>
        {prevTo && (
          <TextLink to={prevTo} rel="prev">
            ← {prevLabel || prevTo}
          </TextLink>
        )}
      </li>
      <li>
        {nextTo && (
          <TextLink to={nextTo} rel="next">
            {nextLabel || nextTo} →
          </TextLink>
        )}
      </li>
    </ul>
  )
}

PostNavigation.propTypes = {
  nextLabel: PropTypes.string,
  nextTo: PropTypes.string,
  prevLabel: PropTypes.string,
  prevTo: PropTypes.string
}
