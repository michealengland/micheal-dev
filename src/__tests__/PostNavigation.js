import React from 'react'
import renderer from 'react-test-renderer'
import PostNavigation from '@components/molecules/PostNavigation'

describe('PostNavigation', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <PostNavigation
          nextLabel="Next Label"
          nextTo="/blog/some-blog-link"
          prevLabel="Prev Label"
          prevTo="/blog/some-blog-link"
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
