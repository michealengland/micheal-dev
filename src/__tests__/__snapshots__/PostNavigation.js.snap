// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostNavigation renders correctly 1`] = `
<nav
  aria-label="Post Navigation"
  className="postNavigation"
  role="navigation"
>
  <ul
    className="inner"
  >
    <li>
      <a
        href="/blog/some-blog-link"
        rel="prev"
      >
        ← 
        Prev Label
      </a>
    </li>
    <li>
      <a
        href="/blog/some-blog-link"
        rel="next"
      >
        Next Label
         →
      </a>
    </li>
  </ul>
</nav>
`;
